---
title: "Analyze"
package: MMRFVariant
#output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{2. Analyze}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---



### <b>MMRFVariant_SurvivalKM function</b>

It performs a Kaplan-Meier (KM) survival analysis that takes into account clinical data and annotated variants form MMRF-RG. Beside of the list of SNPs (<ListSNPs>) to analyze, it requires all three the following datasets:<br>

```{r results = 'hide', message=FALSE, warning=FALSE, eval = F}

patient <- read.csv("~/MMRF_CoMMpass_IA15_PER_PATIENT")
trt <- read.csv("~/MMRF_CoMMpass_IA15_STAND_ALONE_TRTRESP")
variant.ann <- read.csv("~/MMRF_CoMMpass_IA15a_All_Canonical_Variants")


```
<br>
 

### <b>Parameters</b>

**Arguments** | **Description**
-----|-----
patient|is the data.frame of the patient clinical data downloaded from MMRF-Commpass Researcher Gateway 
trt|is the data.frame of the patient clinical data (i.e. treatment-response) downloaded from MMRF-Commpass Researcher Gateway 
variant.ann| is the data.frame of the annotated variants downloaded from MMRF-Commpass Researcher Gateway
Listvariant| is the list of the variants to analyze
[FilterBy](#FilterBy)| Column with groups to plot. This is a mandatory field.
risk.table| show or not the risk table
expand| show or not an expanded plot
legend| Legend title of the figure
xlim| x axis limits e.g. xlim = c(0, 1000). Present narrower X axis, but not affect survival estimates.
main| main title of the plot
labels| labels of the plot
ylab| y axis text of the plot
xlab| x axis text of the plot
filename| The name of the pdf file.
color| Define the colors/Pallete for lines.
width| Image width
height| Image height
pvalue| show p-value of log-rank test
conf.range|  show confidence intervals for point estimates of survival curves.
dpi| Figure quality


### <b>Stratifying the patients cohort</b>

The KM survival curves are drawn with respect to the <a name="FilterBy">FilterBy</a> parameter that can be set for stratifying the patients cohort into the categories shown in the following Table:


**Arguments** | **Description**
-----|-----
Ethnicity | Ethnicity 
Stage | ISS Stage
Treatment |  Treatment class
Bestresp |  Best overall response
Gender |  Female or Male
Effect |  Effect Type of observed mutation
Biotype |  Gene biotype




### Example:

<p align="justify">The following plots show the KM survival curves for each variant (i.e. dbSNP ID) in "ListSNPs_NRAS" (we focus on NRAS gene) with respect to the:<br>

- treatment (i.e. Therapy class)<br>
- Stage <br>
- Bestresp <br>
- Ethnicity.<br>

Note that Only the KM curves with pvalue$<$0.05 are shown.
For simplification purposes, we take into account only the top five SNPs by occurrence.</p>


<!--# TREATMENT-->

```{r results = 'hide', message=FALSE, warning=FALSE, eval = F}
NRAS_SNPs<-head(ListSNPs_NRAS,5) #["rs2157615","rs61731685","rs370560636","rs116293337"]

surv<-MMRFVariant_SurvivalKM(patient,  
                              trt,
                              variant.ann,
                              NRAS_SNPs,
                              FilterBy="Treatment", 
                              filename=NULL,
                              xlim = c(100,3000),
                              conf.range = FALSE,
                              color = c("Dark2"))




```



<div class="panel panel-info">
<div class="panel-heading"><b>Resulting plot</b></div>
<div class="panel-body">
</div>
</div>

```{r figurename2, echo=FALSE, fig.cap="KM survival curves are drawn for each SNP found in the NRAS gene in the case of the FilterBy parameter is set to Treatment", out.width = '99%'}
knitr::include_graphics("imgs/KM_Surv_treatment.png")
```


<!--# STAGE-->

```{r results = 'hide', message=FALSE, warning=FALSE, eval = F}
NRAS_SNPs<-head(ListSNPs_NRAS,5) #["rs2157615","rs61731685","rs370560636","rs116293337"]

surv<-MMRFVariant_SurvivalKM(patient,  
                              trt,
                              variant.ann,
                              NRAS_SNPs,
                              FilterBy="Stage", 
                              filename=NULL,
                              xlim = c(100,3000),
                              conf.range = FALSE,
                              color = c("Dark2"))




```



<div class="panel panel-info">
<div class="panel-heading"><b>Resulting plot</b></div>
<div class="panel-body">
</div>
</div>

```{r figurename3, echo=FALSE, fig.cap="KM survival curves are drawn for each SNP found in the NRAS gene in the case of the FilterBy parameter is set to Stage", out.width = '50%'}
knitr::include_graphics("imgs/KM_Surv_stage.png")
```



<!--# ETHNICITY-->

```{r results = 'hide', message=FALSE, warning=FALSE, eval = F}
NRAS_SNPs<-head(ListSNPs_NRAS,5) #["rs2157615","rs61731685","rs370560636","rs116293337"]

surv<-MMRFVariant_SurvivalKM(patient,  
                              trt,
                              variant.ann,
                              NRAS_SNPs,
                              FilterBy="Ethnicity", 
                              filename=NULL,
                              xlim = c(100,3000),
                              conf.range = FALSE,
                              color = c("Dark2"))




```



<div class="panel panel-info">
<div class="panel-heading"><b>Resulting plot</b></div>
<div class="panel-body">
</div>
</div>

```{r figurename4, echo=FALSE, fig.cap="KM survival curves are drawn for each SNP found in the NRAS gene in the case of the FilterBy parameter is set to Ethnicity", out.width = '99%'}
knitr::include_graphics("imgs/KM_Surv_ethnicity.png")
```





<!--# BESTRESP-->

```{r results = 'hide', message=FALSE, warning=FALSE, eval = F}
NRAS_SNPs<-head(ListSNPs_NRAS,5) #["rs2157615","rs61731685","rs370560636","rs116293337"]

surv<-MMRFVariant_SurvivalKM(patient,  
                              trt,
                              variant.ann,
                              NRAS_SNPs,
                              FilterBy="Bestresp", 
                              filename=NULL,
                              xlim = c(100,3000),
                              conf.range = FALSE,
                              color = c("Dark2"))




```



<div class="panel panel-info">
<div class="panel-heading"><b>Resulting plot</b></div>
<div class="panel-body">
</div>
</div>

```{r figurename5, echo=FALSE, fig.cap="KM survival curves are drawn for each SNP found in the NRAS gene in the case of the FilterBy parameter is set to Bestresp", out.width = '99%'}
knitr::include_graphics("imgs/KM_Surv_Bestresp.png")
```



