---
title: "MMRFVariant - Introduction"

package: MMRFVariant
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{1. Introduction}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  warning=FALSE, 
  message=FALSE,
  comment = "#>"
)
```



```{r setup}
library(MMRFVariant)
library(dplyr)
library(DT)
library(ggplot2)
library(stringr)
library(ggpubr)
library(survminer)
library(survival)
library(formattable)
```


```{r, echo = FALSE,hide=TRUE, message=FALSE,warning=FALSE}
devtools::load_all(".")
```

```{r message=FALSE, warning=FALSE, include=FALSE}
library(dplyr)
library(DT)
library(ggplot2)
library(stringr)
library(ggpubr)
```

## Introduction
<div class="panel panel-info">
<div class="panel-body">
<p align="justify">
MMRF-CoMMpass data can be retrieved directly from [MMRF-Commpass Researcher Gateway](https://research.themmrf.org/). They can be downloaded and imported as a dataframe into R environment.
</p>

</div>
</div>

### Fetching and importing MMRF-COMMPASS datasets

<p align="justify">
In this example we will fetch the MMRF-CoMMpass data from MMRF-COMMPASS Research Gateway (MMRF-RG). Once the dataset (.csv files) is imported into our own R environment, it is can be explored  as a R-dataframe.
</p>

```{r results = 'hide', message=FALSE, warning=FALSE, eval = F}

patient <- read.csv("~/MMRF_CoMMpass_IA15_PER_PATIENT")
trt <- read.csv("~/MMRF_CoMMpass_IA15_STAND_ALONE_TRTRESP")
variant.ann <- read.csv("~/MMRF_CoMMpass_IA15a_All_Canonical_Variants")


```


### MMRFVariant functions

<p align="justify">
The table below shows an overview of the MMRFVariant functions and their outputs.
</p>

**Function** | **Output**
-----|-----
MMRFVariant_PlotbyEffect_Impact | N# of variants classified by Impact and Effect  
MMRFVariant_PlotVariantsbyGene | Heatmap of the N# of variants  
MMRFVariant_GetVariantsbyGene |  list of SNPs occurring in the gene set
MMRFVariant_SurvivalKM |  KM Survival curves
MMRFVariant_getImpact |  Impact table of SNPs

### MMRFVariant Use-Case
For a more intuitive interpretation of the results, MMRFVariant provides a graphical or tabular representation of the outputs.
The use-case diagram shown below provides an overview of the MMRFVariant functionalities and the different scenarios that can be performed.

```{r figurename2, echo=FALSE, fig.cap="KM Survival curves ", out.width = '99%'}
knitr::include_graphics("imgs/UseCase.png")
```



### fictitious datasets useful to test MMRFVariant functions
  


<p align="justify">
In order to test the MMRFVariant functions (without fetch MMRF-CoMMpass data from MMRF-RG), three fictitious datasets are provided below:<br>

- <variant.ann.example> <br>
- <patient.example> <br>
- <trt.example> <br>

They replicate the original datasets format icluding only some fileds. To reproduce this example you should comment the previous code that fetches the MMRF-CoMMpass datasets (.csv files) from MMRF-COMMPASS Research Gateway. 
</p>

<div class="panel panel-info">
<div class="panel-heading"><b>Annotated variant dataframe</b></div>
<div class="panel-body">
</div>
</div>


```{r  echo=TRUE, message=FALSE, warning=FALSE}
datatable(variant.ann.example,options = list(scrollX = TRUE, keys = TRUE))

```

<div class="panel panel-info">
<div class="panel-heading"><b>Patient dataframe</b></div>
<div class="panel-body">
</div>
</div>
```{r  echo=TRUE, message=FALSE, warning=FALSE}
datatable(patient.example,options = list(scrollX = TRUE, keys = TRUE))

```


<div class="panel panel-info">
<div class="panel-heading"><b>Treatment dataframe</b></div>
<div class="panel-body">
</div>
</div>
```{r  echo=TRUE, message=FALSE, warning=FALSE}
datatable(trt.example,options = list(scrollX = TRUE, keys = TRUE))

```

# Session Information
******
```{r sessionInfo}
sessionInfo()
```





